src/ert/scheduler/job.py

```diff
@@ -156,13 +134,37 @@ async def __call__(
         for attempt in range(max_submit):
             await self._submit_and_run_once(sem)
 
-            if self.returncode.cancelled() or (
-                self.returncode.done() and self.returncode.result() == 0
-            ):
+            if self.returncode.cancelled():
+                break
+
+            if self.returncode.result() == 0:
```

Comment:
Maybe we can split all/some of these if statements into separate functions? It is a bit overwhelming to read.

Reply:
Good suggestion, I made a new `_handle_finished_forward_model` that I think made it better.

Reply:
Yes, now it looks great!